src/Orchard.Web/Modules/Orchard.Azure/Services/FileSystems/Media/AzureBlobStorageProvider.cs

```diff
@@ -64,8 +65,10 @@ public AzureBlobStorageProvider(string storageConnectionString, string container
         /// <returns>The corresponding local path.</returns>
         public string GetStoragePath(string url) {
             EnsureInitialized();
-            if (url.StartsWith(_absoluteRoot)) {
-                return HttpUtility.UrlDecode(url.Substring(Combine(_absoluteRoot, "/").Length));
+            var rootUri = new Uri(_absoluteRoot);
+            var uri = new Uri(url);
+            if((uri.Host == rootUri.Host || (!string.IsNullOrWhiteSpace(_publicHostName) && uri.Host == _publicHostName)) && uri.AbsolutePath.StartsWith(rootUri.AbsolutePath)) {
```

Comment:
I have learned that if you don't rely on white spaces, you can use `String.IsNullOrEmpty` only as whitespace comparison is expensive.